compiler error with the latest develop branch

```
In file included from /home/luotao02/Paddle/paddle/platform/dynload/nccl.h:20:0,
                 from /home/luotao02/Paddle/paddle/platform/enforce.h:39,
                 from /home/luotao02/Paddle/paddle/platform/gpu_info.cc:19:
/home/luotao02/Paddle/paddle/platform/call_once.h: In lambda function:
/home/luotao02/Paddle/paddle/platform/call_once.h:36:9: error: parameter packs not expanded with ‘...’:
       f(args...);
         ^
/home/luotao02/Paddle/paddle/platform/call_once.h:36:9: note:         ‘args’
/home/luotao02/Paddle/paddle/platform/call_once.h:36:13: error: expansion pattern ‘args’ contains no argument packs
       f(args...);
             ^
make[2]: *** [paddle/platform/CMakeFiles/gpu_info.dir/gpu_info.cc.o] Error 1
make[1]: *** [paddle/platform/CMakeFiles/gpu_info.dir/all] Error 2
make: *** [all] Error 2
```

--- a/paddle/platform/call_once.h
+++ b/paddle/platform/call_once.h
@@ -27,20 +27,22 @@ namespace platform {
 
  This wrap is a hack to avoid this bug.
 */
-template <class Callable, class... Args>
+template <typename Callable, typename... Args>
 inline void call_once(std::once_flag& flag, Callable&& f, Args&&... args) {
   bool good = false;
   std::exception ex;
-  std::call_once(flag, [&]() {
-    try {
-      f(args...);
-      good = true;
-    } catch (const std::exception& e) {
-      ex = e;
-    } catch (...) {
-      ex = std::runtime_error("excption caught in call_once");
-    }
-  });
+  std::call_once(flag,
+                 [&](Args&&... args) {
+                   try {
+                     f(args...);
+                     good = true;
+                   } catch (const std::exception& e) {
+                     ex = e;
+                   } catch (...) {
+                     ex = std::runtime_error("excption caught in call_once");
+                   }
+                 },
+                 args...);
   if (!good) {
     throw std::exception(ex);
   }

